helper/grpc-middleware/logging/options.go

```diff
@@ -0,0 +1,89 @@
+package logging
+
+import (
+	"github.com/hashicorp/go-hclog"
+	"google.golang.org/grpc/codes"
+)
+
+type options struct {
+	levelFunc CodeToLevel
+}
+
+var defaultOptions = &options{}
+
+type Option func(*options)
+
+func evaluateClientOpt(opts []Option) *options {
+	optCopy := &options{}
+	*optCopy = *defaultOptions
+	optCopy.levelFunc = DefaultCodeToLevel
+	for _, o := range opts {
+		o(optCopy)
+	}
+	return optCopy
+}
+
+func WithStatusCodeToLevelFunc(fn CodeToLevel) Option {
+	return func(opts *options) {
+		opts.levelFunc = fn
+	}
+}
+
+// CodeToLevel function defines the mapping between gRPC return codes and hclog level.
+type CodeToLevel func(code codes.Code) hclog.Level
+
+func DefaultCodeToLevel(code codes.Code) hclog.Level {
+	switch code {
+	// Trace Logs -- Useful for Nomad developers but not necessarily always wanted
+	case codes.OK:
+		return hclog.Trace
+
+	// Debug logs
+	case codes.Canceled:
+		return hclog.Debug
+	case codes.InvalidArgument:
+		return hclog.Debug
+	case codes.ResourceExhausted:
+		return hclog.Debug
+	case codes.FailedPrecondition:
+		return hclog.Debug
+	case codes.Aborted:
+		return hclog.Debug
+	case codes.OutOfRange:
+		return hclog.Debug
+	case codes.NotFound:
+		return hclog.Debug
+	case codes.AlreadyExists:
+		return hclog.Debug
+
+	// Info Logs - More curious/interesting than debug, but not necessarily critical
+	case codes.Unknown:
+		return hclog.Info
+	case codes.DeadlineExceeded:
+		return hclog.Info
+	case codes.PermissionDenied:
+		return hclog.Info
+	case codes.Unauthenticated:
+		// unauthenticated requests are probably usually fine?
```

Comment:
So right now I'm not sure when we'd actually receive `Unauthenticated` responses, so bc we won't have explicit handling for them I'd rather surface them to operators in case of Weirdness (like what if a plugin decides to return one when their auth keys for the service provider expire?)